src/scenes/AcceleratorRouter/Documents/DocumentView/DocumentOutlinePanel.tsx

```diff
@@ -0,0 +1,239 @@
+} from '@mui/material';
+
+import Icon from 'src/components/common/icon/Icon';
+import { useIsVisible } from 'src/hooks/useIsVisible';
+import { requestListVariablesValues } from 'src/redux/features/documentProducer/values/valuesThunks';
+import { selectVariablesWithValues } from 'src/redux/features/documentProducer/variables/variablesSelector';
+import { requestListVariables } from 'src/redux/features/documentProducer/variables/variablesThunks';
+import { useSelectorProcessor } from 'src/redux/hooks/useSelectorProcessor';
+import { useAppDispatch, useAppSelector } from 'src/redux/store';
+import { Document as DocumentType } from 'src/types/documentProducer/Document';
+import { SectionVariableWithValues, VariableWithValues } from 'src/types/documentProducer/Variable';
+
+const isHTMLElementInViewport = (element: HTMLElement): boolean => {
+  const rect = element.getBoundingClientRect();
+
+  return (
+    rect.top >= 0 &&
+    rect.left >= 0 &&
+    rect.bottom <= (window.innerHeight || document.documentElement.clientHeight) &&
+    rect.right <= (window.innerWidth || document.documentElement.clientWidth)
+  );
+};
+
+const StyledTooltip = styled(({ className, ...props }: TooltipProps) => (
+  <Tooltip {...props} arrow classes={{ popper: className }} />
+))(({ theme }) => ({
+  [`& .${tooltipClasses.arrow}`]: {
+    color: theme.palette.TwClrBgInverse,
+  },
+  [`& .${tooltipClasses.tooltip}`]: {
+    backgroundColor: theme.palette.TwClrBgInverse,
+  },
+}));
+
+type SectionItemProps = {
+  ancestorSectionNumbers?: string[];
+  section: SectionVariableWithValues;
+  selectedSectionId: number | undefined;
+  setSelectedSectionId: React.Dispatch<React.SetStateAction<number | undefined>>;
+};
+
+const SectionItem = ({
+  ancestorSectionNumbers = [],
+  section,
+  selectedSectionId,
+  setSelectedSectionId,
+}: SectionItemProps): JSX.Element => {
+  const theme = useTheme();
+  const childSections = section.children.filter((child) => child.type === 'Section') as SectionVariableWithValues[];
+  const label = `${section.sectionNumber}${section.parentSectionNumber ? '' : '.'} ${section.name}`;
+  const isSelected = section.id === selectedSectionId;
+
+  const onClick = useCallback(() => {
+    // do nothing if already selected
+    if (section.id === selectedSectionId) {
+      return;
+    }
+
+    // update selected section id
+    setSelectedSectionId(section.id);
+
+    // exit if section has no section number
+    if (!section.sectionNumber) {
+      return;
+    }
+
+    // get document heading element
+    const element = document.getElementById(section.sectionNumber);
+    if (!element) {
+      // exit if element not found
+      return;
+    }
+
+    if (!isHTMLElementInViewport(element)) {
+      // scroll to section heading in the document, if not already in view
+      element.scrollIntoView({ behavior: 'smooth' });
+    }
+  }, [section.id, selectedSectionId, setSelectedSectionId]);
+
+  return (
+    <Box>
+      <ButtonBase
```

Comment:
Would the web-components <Link> component work here? It has an onClick prop

Reply:
I think my initial thoughts were that it wouldn't work for some reason, but perhaps it will. Good suggestion, will give it a try.

Reply:
It seems that our `Button` component in `web-components` doesn't accept `children`, so it won't work for this case.